src/transport/types.ts

```diff
@@ -0,0 +1,120 @@
+import events from 'events'
+
+/**
+ * A libp2p transport is understood as something that offers a dial and listen interface to establish connections.
+ */
+export interface Interface {
```

Comment:
More concrete name would be much better here.

Reply:
removing the class, I will be calling Transport